### PR TITLE
Implement feedback from `nucypher` integration

### PR DESCRIPTION
**Pull request by piotr-roslaniec**
**To Do:**
- [x] Write a changelog entry for `ferveo` integration

**Type of PR:**
- Feature

**Required reviews:** 
- 2

**What this does:**
- Implements issues caught during integration with `nucypher`

**Issues fixed/closed:**
> - Fixes #...

**Why it's needed:**
- Discovering and addressing changes before signing off on the new `nucypher-core` version
- Finalizing `nucypher` integration

**Notes for reviewers:**
- Please consider reviewing https://github.com/nucypher/nucypher/pull/3135 first

### CODE REVIEW

File: nucypher-core-python/Cargo.toml [hunk 1]
```diff
@@ -11,8 +11,8 @@ crate-type = ["cdylib"]
 pyo3 = "0.18"
 nucypher-core = { path = "../nucypher-core" }
 # TODO: Replace with a new release after this PR is merged and released: https://github.com/nucypher/rust-umbral/pull/122
-umbral-pre = { git = "https://github.com/piotr-roslaniec/rust-umbral.git", rev = "90a5075bb3462421fa6db79463960cd8e10a7025", features = ["bindings-python"] }
-ferveo = { package = "ferveo-pre-release", version = "0.1.0-alpha.7", features = ["bindings-python"] }
+umbral-pre = { git = "https://github.com/piotr-roslaniec/rust-umbral.git", rev = "5a279039d1e44e415f2cfac85282fc5102d41a6c", features = ["bindings-python"] }
+ferveo = { package = "ferveo-pre-release", git = "https://github.com/nucypher/ferveo.git", rev = "9795e2969f525fc7ee3a7721dca368287f4825c8", features = ["bindings-python"] }
```
Comment (fjarri) [inline nucypher-core-python/Cargo.toml:15]: Should be pinned to a released version before merge.

File: nucypher-core-wasm/Cargo.toml [hunk 2]
```diff
@@ -20,14 +20,15 @@ default = ["console_error_panic_hook"]
...
+umbral-pre = { git = "https://github.com/piotr-roslaniec/rust-umbral.git", rev = "5a279039d1e44e415f2cfac85282fc5102d41a6c", features = ["bindings-wasm"] }
+ferveo = { package = "ferveo-pre-release", git = "https://github.com/nucypher/ferveo.git", rev = "9795e2969f525fc7ee3a7721dca368287f4825c8", features = ["bindings-wasm"] }
 nucypher-core = { path = "../nucypher-core" }
 wasm-bindgen = "0.2.86"
 js-sys = "0.3.63"
 console_error_panic_hook = { version = "0.1", optional = true }
 derive_more = { version = "0.99", default-features = false, features = ["from", "as_ref"] }
-wasm-bindgen-derive = "0.2.0"
+# TODO: Replace with a new release after this issue is fixed: https://github.com/fjarri/wasm-bindgen-derive/issues/3
+wasm-bindgen-derive = { git = "https://github.com/piotr-roslaniec/wasm-bindgen-derive.git", rev = "d13475a9b36d6c6e5f2f99372d175f21b00bf0d6" }
```
Comment (fjarri) [inline nucypher-core-wasm/Cargo.toml:31]: Does the `[profile.release]` in `wasm-bindgen-derive` really affect the build of `nucypher-core`? https://doc.rust-lang.org/cargo/reference/profiles.html states "Profile settings defined in dependencies will be ignored."

File: nucypher-core-python/nucypher_core/ferveo.pyi [hunk 3]
```diff
@@ -0,0 +1,261 @@
...
+        ...
+
+    public_key: DkgPublicKey
+
+    public_params: DkgPublicParameters
+
+    def generate_transcript(self) -> Transcript:
+        ...
+
+    def aggregate_transcripts(self, messages: Sequence[(Validator, Transcript)]) -> AggregatedTranscript:
```
Comment (fjarri) [inline nucypher-core-python/nucypher_core/ferveo.pyi:87]: Does `(Validator, Transcript)` work with type checkers? I would think you'd have to do `Tuple[Validator, Transcript]`

Comment (fjarri) [reply nucypher-core-python/nucypher_core/ferveo.pyi:87]: (same in `AggregatedTranscript`)

File: nucypher-core-wasm/Cargo.toml (hunk 2 above)
Comment (derekpierre) [reply nucypher-core-wasm/Cargo.toml:31]: This can be updated now to v0.2.1.

File: nucypher-core-wasm/Cargo.toml [hunk 4]
```diff
@@ -20,14 +20,15 @@ default = ["console_error_panic_hook"]
 
 [dependencies]
 # TODO: Replace with a new release after this PR is merged and released: https://github.com/nucypher/rust-umbral/pull/122
-umbral-pre = { git = "https://github.com/piotr-roslaniec/rust-umbral.git", rev = "90a5075bb3462421fa6db79463960cd8e10a7025", features = ["bindings-wasm"] }
-ferveo = { package = "ferveo-pre-release", version = "0.1.0-alpha.7", features = ["bindings-wasm"] }
+umbral-pre = { git = "https://github.com/piotr-roslaniec/rust-umbral.git", rev = "5a279039d1e44e415f2cfac85282fc5102d41a6c", features = ["bindings-wasm"] }
+ferveo = { package = "ferveo-pre-release", git = "https://github.com/nucypher/ferveo.git", rev = "9795e2969f525fc7ee3a7721dca368287f4825c8", features = ["bindings-wasm"] }
```
Comment (derekpierre) [inline nucypher-core-wasm/Cargo.toml:24]: Has testing been successful, and these vcs dependencies are no longer necessary?

File: nucypher-core-python/nucypher_core/ferveo.pyi (hunk 3 above)
Comment (piotr-roslaniec) [reply nucypher-core-python/nucypher_core/ferveo.pyi:87]: It works, but `Tuple[Validator, Transcript]` is better

File: CHANGELOG.md [hunk 5]
```diff
@@ -3,11 +3,27 @@
 The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
+
+## Unreleased
```
Comment (derekpierre) [inline CHANGELOG.md:7]:
```suggestion
## [0.9.0] - Unreleased
```

**Review by derekpierre — APPROVED** [review]
🎸 - see comment in Changelog file. Onced updated, it's good to go.

**Pull request merged**